pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -223,6 +265,17 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
 		}
 		isvc.Status.PropagateStatus(v1beta1.PredictorComponent, status)
 	}
+	statusSpec, _ := isvc.Status.Components[v1beta1.PredictorComponent]
+	if rawDeployment {
+		podLabelValue = constants.GetRawServiceLabel(constants.DefaultPredictorServiceName(isvc.ObjectMeta.Name))
+	} else {
+		podLabelValue = statusSpec.LatestCreatedRevision
+	}
+	podList, err := isvcutils.ListPodsByLabel(p.client, isvc.ObjectMeta.Namespace, podLabelKey, podLabelValue)
+	if err != nil {
+		return errors.Wrapf(err, "fails to list inferenceservice pods by label")
+	}
+	isvc.Status.PropagateModelStatus(statusSpec, podList, rawDeployment)
```

Comment:
I think that's good point! I did not realize that `targetModelState` is not actually populated and we are only setting `activeModelState`. `targetModelState` is equivalent to `lastestCreatedRevision` and `activeModelState` is equivalent to `lastestReadyRevision` in knative mode.

Regarding the spec processing I think there are still dependencies for the resource reconciliations, for example if the deployment reconciliation fails then the controller should not continue to create the ingress resource, otherwise user may not get unexpected error when they call the service. I agree the status checking + update should be separate and happen regardless of the reconciliation result.